ui/components/app/snaps/keyring-snap-removal-warning/keyring-snap-removal-warning.tsx

```diff
@@ -0,0 +1,189 @@
+  const [confirmationInput, setConfirmationInput] = useState('');
+  const [error, setError] = useState(false);
+  const { chainId } = useSelector(getProviderConfig);
+
+  const validateConfirmationInput = (input: string): boolean => {
+    setError(false);
+    if (input === snap.manifest.proposedName) {
+      return true;
+    }
+    setError(true);
+    return false;
+  };
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent
+          modalDialogProps={{
+            display: Display.Flex,
+            flexDirection: FlexDirection.Column,
+            gap: 4,
+          }}
+        >
+          <ModalHeader
+            onBack={() => {
+              setShowConfirmation(false);
+              onBack();
+            }}
+            onClose={() => {
+              setShowConfirmation(false);
+              onClose();
+            }}
+          >
+            {t('removeSnap')}
+          </ModalHeader>
+          {showConfirmation === false ? (
+            <>
+              <BannerAlert severity={BannerAlertSeverity.Warning} className="">
+                {t('backupKeyringSnapReminder')}
+              </BannerAlert>
+              <Box
+                display={Display.Flex}
+                justifyContent={JustifyContent.spaceBetween}
+              >
+                <Text>{t('removeKeyringSnap')}</Text>
+                <InfoTooltip
+                  contentText={t('removeKeyringSnapToolTip')}
+                  position="top"
+                />
+              </Box>
+              {keyringAccounts.map((account, index) => {
+                return (
+                  <KeyringAccountListItem
+                    key={index}
+                    account={account}
+                    snapUrl={getAccountLink(account.address, chainId)}
+                  />
+                );
+              })}
+            </>
+          ) : (
+            <>
+              <Box
+                display={Display.Flex}
+                flexDirection={FlexDirection.Column}
+                marginTop={6}
+              >
+                <BannerAlert
+                  severity={BannerAlertSeverity.Warning}
+                  className=""
+                  marginBottom={4}
+                >
+                  {t('backupKeyringSnapReminder')}
+                </BannerAlert>
+                <Text marginBottom={4}>
+                  {t('keyringSnapRemoveConfirmation', [
+                    <Text
+                      key="keyringSnapRemoveConfirmation2"
+                      fontWeight={FontWeight.Bold}
+                      as="span"
+                    >
+                      {snap.manifest.proposedName}
+                    </Text>,
+                  ])}
+                </Text>
+                {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+                {/* @ts-ignore TODO: fix TextField props */}
```

Comment:
what's missing here?

Reply:
Currently all the TextField props are all required, and we would have to change it to this 

`<TextField
                  marginBottom={4}
                  value={confirmationInput}
                  onChange={(e: { target: { value: string } }) => {
                    setConfirmationInput(e.target.value);
                    setConfirmedRemoval(
                      validateConfirmationInput(e.target.value),
                    );
                  }}
                  error={error}
                  inputProps={{
                    'data-testid': 'remove-snap-confirmation-input',
                  }}
                  type="text"
                  autoComplete={undefined}
                  autoFocus={undefined}
                  className={undefined}
                  defaultValue={undefined}
                  disabled={undefined}
                  id={undefined}
                  inputRef={undefined}
                  startAccessory={undefined}
                  endAccessory={undefined}
                  maxLength={undefined}
                  name={undefined}
                  onBlur={undefined}
                  onClick={undefined}
                  onFocus={undefined}
                  placeholder={undefined}
                  readOnly={undefined}
                  required={undefined}
                  testId={undefined}
                />`

Reply:
Gotcha